### ISSUE TITLE
Make plots responsive only in x or y

### ISSUE THREAD

**Issue by cgmorton** [bokeh/bokeh#2920]
First, the new responsive plots are awesome!  Would it be possible to make them responsive only for one axis though?  Basically, I want to show three vertical plots (using vplot) and have all three plots be visible without scrolling, but also have the plots fill the window horizontally.  

Alternatively, would there be a way to make vplot "responsive" or maybe to set a maximum height for a figure?

The following code should show three responsive plots, but I have to scroll to see the bottom 2 because they are filling horizontally.
```
import os
from bokeh.plotting import figure, output_file, show, vplot
import numpy as np

output_name = 'responsive_vplot_test'
f = output_file(os.path.join(os.getcwd(), output_name+'.html'), title=output_name)
TOOLS = 'xpan,xwheel_zoom,box_zoom,reset,save'

x_array = np.arange(0,10,1)
f1 = figure(tools=TOOLS, toolbar_location="right", responsive=True)
f1.line(x_array, x_array)
f2 = figure(x_range=f1.x_range, tools=TOOLS, toolbar_location="right", responsive=True)
f2.line(x_array, x_array)
f3 = figure(x_range=f1.x_range, tools=TOOLS, toolbar_location="right", responsive=True)
f3.line(x_array, x_array)

show(vplot(f1, f2, f3))
```

**Comment by birdsarah** [comment]
@cgmorton, am I correct in thinking that if we did this, the plots would lose their original aspect ratio?

**Comment by bryevdv** [comment]
An option is to adjust the data bounds during resizing, to keep the aspect constant. That would require a little work.

**Comment by cgmorton** [comment]
Yes, in order to fill the window, I'm picturing the aspect ratio changing as the user resizes the window.  I am picturing something similar to the example provided in https://github.com/bokeh/bokeh/pull/2950, but each frame or box would be a separate plot.

**Comment by birdsarah** [comment]
Fixed in #4343
